src/core_plugins/timelion/public/panels/timechart/schema.js

```diff
@@ -274,6 +281,11 @@ export default function timechartFn(Private, config, $rootScope, timefilter, $co
           _.each(canvasElem.find('.ngLegendValue'), function (elem) {
             $compile(elem)(legendScope);
           });
+
+          if (_.get($scope.plot.getData(), '[0]._global.legend.showTime', true)) {
```

Comment:
fwiw, we can probably remove the double normalization: 
`_.get($scope.plot.getData(), '[0]._global.legend.showTime')`

since we already do that in the `legend`-function implementation

Reply:
Its still needed because the legend function may not be used in the expression. In that case, _global.legend is `undefined`

Reply:
ow ok, makes sense, thx!